### DISCUSSION TITLE
Future tasks

### DISCUSSION THREAD

**Discussion by jaker-dotcom** [GothenburgBitFactory/taskwarrior#2525] [Q&A]
While I am at it: Not only tasks that could have long been archived are clogging up the search: I sometimes refrain from entering a task just yet because it inflates the IDs of tasks. 

The approach I was thinking of before but just didn't have the knowledge to realize is a cronjob that performs task add on entries in a file with tasks that can be entered at a later date so to not show up in tasksh every now and then and to not inflate the IDs. 

But these tasks - contrary to archived tasks - would ideally be searchable. Which of course they are but through other means than taskwarriors search. Also "entered date" would be untrue. Is there a way to setup taskrc such that tasks waiting for a long time don't get assigned an ID. 

It's not too important for me but I'd rather just type two numbers instead of three when performing commands on tasks.

What's your take on this?
Jaker

Edit: Typing this I realize that another UDA would probably suffice at least for the unneccesary tasksh reviews. Those really turn me off either from entering future tasks or using tasksh.

**Comment by tbabej** [comment]
The only way of adding new tasks that satisfies both of your criteria, that is:
a) not showing up in tasksh review
b) not inflating IDs

is to add the task as already in a non-pending state (i.e. deleted) with a specific tag so that you can "un-delete" it later once you deem it ready:

```
[warrior@tw-2.6.0 ~]$ task add This should be in the list
Created task 1.

[warrior@tw-2.6.0 ~]$ task add +future status:deleted I don\'t want this to show up
Created task 0e054ba2-1bd7-4cec-be37-6e4778bf6200.

[warrior@tw-2.6.0 ~]$ task add This task does not have ID inflated
Created task 2.

[warrior@tw-2.6.0 ~]$ # later, when I want to make my future tasks pending again
[warrior@tw-2.6.0 ~]$ task status:deleted +future mod status:pending
Modifying task 0e054ba2 'I don't want this to show up'.
Modified 1 task.

[warrior@tw-2.6.0 ~]$ task
ID Age   Tag    Description                         Urg 
 - 43s   future I don't want this to show up         0.8
 1  1min        This should be in the list             0
 2 24s          This task does not have ID inflated    0

3 tasks
```

**Reply by jaker-dotcom** [reply]
You mastermind :D
Thanks! 
Just have to come up with an idea to make them pending at the right time (so that it can be specified upon entry). Maybe a fake schedule date that so that the task will be rendered pending by a cronjob when the schedule date is within e.g. two weeks. Upon making it pending the schedule date get's removed. 

I'll try to work with that for a while.
